Use cobra to parse cmdline args - phase 1

zivnevo opened the pull request:
Here we just replace flags with cobra to fill `InArgs` and do all the checks. Then we call the functions in `main` (almost) as usual.
Phase 2 should be getting rid of InArgs, and call pkg functions directly.

Addresses: #537

adisos reviewed (approved):
Looks good!
Should update the `Readme.md` file (in this PR or the next one)

zivnevo commented:
> Looks good! Should update the `Readme.md` file (in this PR or the next one)

Updated README. It now spans over several pages (one main page + one page per command).
Also added run examples and documented output formats.
Please take a look

adisos reviewed (approved):
looks good!
